dmutils/direct_plus_client.py

```diff
@@ -95,12 +107,34 @@ def get_organization_by_duns_number(self, duns_number) -> Optional[dict]:
             response.raise_for_status()
         except HTTPError as exception:
             try:
-                error = response.json()['error']
+                error = response.json()["error"]
+            except (ValueError, KeyError) as e:
+                error_message = f"Unable to parse DirectPlus API response: {response}: {e}"
+                raise DirectPlusError(error_message, exception)
 
-                if (response.status_code, error["errorCode"]) in [DUNS_NUMBER_INVALID, DUNS_NUMBER_NOT_FOUND]:
-                    return None
+            error_message = (
+                f"Unable to get supplier by DUNS number (HTTP error {response.status_code}): {error}"
+            )
+
+            if response.status_code in [DUNSNumberInvalid.status_code, DUNSNumberNotFound.status_code]:
+                self.logger.warning(error_message)
+
+                if response.status_code == DUNSNumberInvalid.status_code:
+                    raise DUNSNumberInvalid(error_message, exception)
+                if response.status_code == DUNSNumberNotFound.status_code:
+                    raise DUNSNumberNotFound(error_message, exception)
+
+            else:
+                self.logger.error(error_message)
+                raise DirectPlusError(error_message, exception)
+
+        try:
+            organization = cast(dict, response.json()['organization'])
+        except (ValueError, KeyError) as e:
+            # this should never happen, so let's propogate it so it is noisy
+            self.logger.error(
+                f"Unable to parse DirectPlus API response (HTTP code {response.status_code}): {response}"
+            )
+            raise e
```

Comment:
Python nit: if you're just re-raising `e`, you don't need to explicitly catch it - https://docs.python.org/3/tutorial/errors.html#raising-exceptions

Reply:
Oh, thanks for the nit, I meant to log the `e` not just reraise it